isis/src/lro/apps/lrowac2isis/lrowac2isis.cpp

```diff
@@ -0,0 +1,833 @@
+
+      uveven = new Cube();
+      uvodd  = new Cube();
+
+      // 4 output lines (1 framelet) from 2uv lines
+      uvOutputLineRatio = 4.0 / (4.0 * 2.0);
+    }
+    else if(instModeId == "BW") {
+      numFilters = 1;
+
+      frameletLines.push_back(14);
+
+      viseven = new Cube();
+      visodd  = new Cube();
+    }
+
+    padding.resize(numFilters);
+
+
+    for(int filter = 0; filter < numFilters; filter++) {
+      padding[filter] = (colorOffset * frameletLines[filter]) * filter;
+
+      // dont count UV for VIS offsetting
+      if(instModeId == "COLOR" && filter > 1) {
+        padding[filter] -= 2 * colorOffset * frameletLines[filter];
+      }
+    }
+
+    FileName baseFileName(ui.GetFileName("TO"));
+
+    if(uveven && uvodd) {
+      // padding[1] is max padding for UV
+      int numSamples = ((viseven) ? p.Samples() / 4 : p.Samples());
+      numSamples     = 128; // UV is alway sum 4 so it is 128 samples
+      int numLines   = (int)(uvOutputLineRatio * inputCubeLines + 0.5) + padding[1];
+      int numBands   = 2;
+
+      uveven->setDimensions(numSamples, numLines, numBands);
+      uveven->setPixelType(Isis::Real);
+
+      QString filename = baseFileName.path() + "/" + baseFileName.baseName() + ".uv.even.cub";
+      uveven->create(filename);
+
+      uvodd->setDimensions(numSamples, numLines, numBands);
+      uvodd->setPixelType(Isis::Real);
+
+      filename = baseFileName.path() + "/" + baseFileName.baseName() + ".uv.odd.cub";
+      uvodd->create(filename);
+    }
+
+    if(viseven && visodd) {
+      // padding[size-1] is max padding for vis (padding[0] or padding[4] or padding[6])
+      int numSamples = p.Samples();
+      int numLines   = (int)(visOutputLineRatio * inputCubeLines + 0.5) + padding[padding.size()-1];
+      int numBands   = ((uveven) ? padding.size() - 2 : padding.size());
+
+      viseven->setDimensions(numSamples, numLines, numBands);
+      viseven->setPixelType(Isis::Real);
+
+      QString filename = baseFileName.path() + "/" + baseFileName.baseName() + ".vis.even.cub";
+      viseven->create(filename);
+
+      visodd->setDimensions(numSamples, numLines, numBands);
+      visodd->setPixelType(Isis::Real);
+
+      filename = baseFileName.path() + "/" + baseFileName.baseName() + ".vis.odd.cub";
+      visodd->create(filename);
+    }
+
+    Pvl isis3VisEvenLab, isis3VisOddLab, isis3UvEvenLab, isis3UvOddLab;
+
+    TranslateLabels(pdsLab, isis3VisEvenLab, isis3VisOddLab, isis3UvEvenLab, isis3UvOddLab, ui);
+
+    writeNullsToFile();
+    p.StartProcess(separateFramelets);
+    p.EndProcess();
+
+    // Add original labels
+    OriginalLabel origLabel(pdsLab);
+
+    int numFramelets = padding.size();
+    PvlKeyword numFrameletsKeyword("NumFramelets", toString(numFramelets));
+
+    if(uveven) {
+      for(int grp = 0; grp < isis3UvEvenLab.groups(); grp++) {
+        uveven->putGroup(isis3UvEvenLab.group(grp));
+      }
+
+      History history("IsisCube");
+      uveven->write(history);
```

Comment:
In the original main.cpp there was a line here with `history.AddEntry();` between 232 and 233 in which causes a segfault. I tried to find an alternative way of including this line without causing a segfault but could not find a solution. Regardless of the segfault, the app would still output the correct files and when I excluded this line, the files would still output correctly. I would appreciate some suggestions of how to fix this segfault with adding back `history.AddEntry();` or if I could get confirmation that it is okay to remove this line.

Reply:
Something I have done in the past is wrap in an if statement: 

```C++
    // Add History
    if (iApp) {
        History history("IsisCube");
        history.AddEntry();
        g_ocube->write(history);
    }
```

This will only run if ran from the command line. The reason it segfaults is because the way the History object works it needs iApp instance (which is a singleton that is only non-null when running from the command line). We definitely need to make the History add stuff to work outside a command line settings, but this I think is a decent fix for now. It'll still add history, just only when running the function as an ISIS app.